tests/unittests/OperatorTest.cpp

```diff
@@ -3504,6 +3645,133 @@ TEST_P(OperatorTest, NonSquareKernelConvolution) {
+      F_->createConv3D("Conv3D", input, filter, zeroBias, outTy, {1, 2, 3},
+                       {1, 1, 1}, {0, 0, 0, 0, 0, 0}, 1);
+  SaveNode *S = F_->createSave("save", CN);
+  bindings_.allocate(S->getPlaceholder());
+
+  ::glow::convertPlaceholdersToConstants(F_, bindings_,
+                                         {input, S->getPlaceholder()});
+  EE_.compile(CompilationMode::Infer, F_);
+  EE_.run(bindings_);
+  Tensor &result = *bindings_.get(S->getPlaceholder());
+
+  static const float ref[] = {106, 127, 190,  211,  274,  295,  442,  463,
+                              526, 547, 610,  631,  778,  799,  862,  883,
+                              946, 967, 1114, 1135, 1198, 1219, 1282, 1303};
+  for (size_t i = 0; i < 4 * 3 * 2; i++) {
+    EXPECT_EQ(result.getHandle().raw(i), ref[i]);
+  }
+}
+
+/// Check Non-cubic kernel for conv3D with quantized input, filters, and bias.
+TEST_P(OperatorTest, NonCubicKernelConv3DQuantized) {
+  ENABLED_BACKENDS(Interpreter);
+
+  auto *input = mod_.createPlaceholder(ElemKind::FloatTy, {1, 4, 4, 4, 1},
+                                       "input", false);
+  auto IH = bindings_.allocate(input)->getHandle();
+  int nextVal = 1;
+  for (size_t i = 0; i < 4; i++) {
+    for (size_t j = 0; j < 4; j++) {
+      for (size_t k = 0; k < 4; k++) {
+        IH.at({0, i, j, k, 0}) = static_cast<float>(nextVal++);
+      } // D
+    }   // W
+  }     // H
+
+  auto qInType = mod_.uniqueType(ElemKind::Int16QTy, {1, 4, 4, 4, 1}, 0.1, 0);
+  auto *qInput = F_->createQuantize("q_input", input, qInType);
+
+  auto filter = mod_.createPlaceholder(ElemKind::FloatTy, {1, 1, 2, 3, 1},
+                                       "filter", false);
+  auto FH = bindings_.allocate(filter)->getHandle();
+  nextVal = 1;
+  for (size_t i = 0; i < 1; i++) {
+    for (size_t j = 0; j < 2; j++) {
+      for (size_t k = 0; k < 3; k++) {
+        FH.at({0, i, j, k, 0}) = static_cast<float>(nextVal++);
+      } // D
+    }   // W
+  }     // H
+
+  auto qFilterType =
+      mod_.uniqueType(ElemKind::Int16QTy, {1, 1, 2, 3, 1}, 0.1, 0);
+  auto *qFilter = F_->createQuantize("q_filter", filter, qFilterType);
+
+  auto *bias = mod_.createPlaceholder(ElemKind::FloatTy, {1}, "bias", false);
+  bindings_.allocate(bias)->zero();
+
+  auto qBiasType = mod_.uniqueType(ElemKind::Int32QTy, {1}, 0.1, 0);
+  auto *qBias = F_->createQuantize("q_bias", bias, qBiasType);
+
+  auto outTy = mod_.uniqueType(ElemKind::FloatTy, {1, 4, 3, 2, 1});
+
+  Convolution3DNode *CN =
+      F_->createConv3D("Conv3D", input, filter, bias, outTy, {1, 2, 3},
+                       {1, 1, 1}, {0, 0, 0, 0, 0, 0}, 1);
+
+  auto qOutTy = mod_.uniqueType(ElemKind::Int16QTy, {1, 4, 3, 2, 1}, 0.1, 0);
+
+  Convolution3DNode *qCN =
+      F_->createConv3D("q_Conv3D", qInput, qFilter, qBias, qOutTy, {1, 2, 3},
+                       {1, 1, 1}, {0, 0, 0, 0, 0, 0}, 1);
+
+  SaveNode *S = F_->createSave("save", CN);
+
+  auto *deQ = F_->createDequantize("deQ_result", qCN);
```

Comment:
minor nit: Since you're using explicit class names for some of the other nodes in these tests, would be nice to be consistent and not use `auto` here.